test/TaskControllers.test.ts

```diff
@@ -0,0 +1,79 @@
+import { Database } from "../src/supabase/supabase";
+import { addTask } from "../src/controllers/AddTaskController"
+import { updateTask } from "../src/controllers/UpdateTaskController"
+import { test_manualCreateProperty, test_manualCreateRoom, test_manualCreateTask, test_manualDeleteProperty, test_manualDeleteTask } from "./TestUtil";
+
+describe('tasks', () => {
+    let property: Database['public']['Tables']['Properties']['Row'];
+    let room: Database['public']['Tables']['Rooms']['Row'];
+
+    beforeAll(async () => {
+        // create temp property and room for tasks to correspond to
+        property = await test_manualCreateProperty();
+        room = await test_manualCreateRoom(property.property_id);
+    })
+
+    afterAll(async () => {
+        // delete temp property cascades to delete room too
+        await test_manualDeleteProperty(property.property_id);
+    })
+
+    describe('adding a task', () => {
+        test('should create a task with a unique task id', async () => {
+            // create object test task to add to database
+            // task_id must be specified so we know that function returns correct ID
+            const task: Database['public']['Tables']['Tasks']['Insert'] = {
```

Comment:
FYI you don't have to specify a type here. It's useful for reading, though addTask should accept the insert with or without this explicit type.